File: command_tree.js

import * as re from "./command_parser_js/RegExp.js";
import * as BaseMatch from "./command_parser_js/base_match_class.js";
import * as SpecialMatch from "./command_parser_js/special_match.js";
import {Command_Parser} from "./command_parser_js/parser_system.js";


const command_ability = new SpecialMatch.Command_Root()
command_ability.add_leaves(
    new BaseMatch.Char("ability").add_leaves(
        ...SpecialMatch.BE_Selector_Tree(
            new BaseMatch.Enum("worldbuilder","mayfly","mute").add_leaves(
                new BaseMatch.Enum("true","false").add_leaves( new BaseMatch.End_Tag() ),
                new BaseMatch.End_Tag()
            )
        )
    ),
    new BaseMatch.Char("say").add_leaves( new BaseMatch.AnyString() ),
    new BaseMatch.Char("execute"),
)
command_ability.tree_leaves[2].add_leaves(
    ...SpecialMatch.BE_Selector_Tree(
        ...SpecialMatch.Pos_Tree(
            ...command_ability.tree_leaves
        )
    )
)



const str1 = 'execute @a[tag="abc \\",d=e] ~~~ say hello " ] ^^121^123 execute @s 123~123 123 say world'
const a = new Command_Parser(command_ability)

console.log([str1]);
console.log(re.compile("(\\^)[-\\+]?[0-9\\.]{0,}").match("^+11"))
try { console.log( a.parser(str1) ) ; console.log( a.Token_list ) }
catch (e) { 
    throw e
    console.log( e.pos )
    console.log( e.word )
    console.log( a.Token_list ) 
}

File: command_tree.py
from command_parser import BaseMatch,SpecialMatch,ParserSystem
import time

command_ability = SpecialMatch.Command_Root().add_leaves(
    BaseMatch.Char("Command:","ability").add_leaves(
        *SpecialMatch.BE_Selector_Tree(
            BaseMatch.Enum("Ability_Argument","worldbuilder","mayfly","mute").add_leaves(
                BaseMatch.Enum("Value","true","false").add_leaves( BaseMatch.End_Tag() ),
                BaseMatch.End_Tag()
            )
        )
    ),
    BaseMatch.Char("Command","say").add_leaves( BaseMatch.AnyMsg("Say_Msg") ),
    BaseMatch.Char("Command","execute"),
)
command_ability.tree_leaves[2].add_leaves(
    *SpecialMatch.BE_Selector_Tree(
        *SpecialMatch.Pos_Tree(
            *command_ability.tree_leaves
        )
    )
)



str1 = 'ability   @e  [rm=-1.,scores={"1"=1, "2"=..1, "3"=1.., "4"=1..1, "5"=!1, "6"=!..1, "7"=!1.., "8"=!1..1}]'
a = ParserSystem.Command_Parser(command_ability)
t1 = time.time()
for i in range(1) : print(a.parser(str1))
print(a.Token_list)
print(time.time() - t1)
exit()
print(
  a.parser(str1)[0].word, "\n",
  a.parser(str1)[0].pos, "\n",
)
print(str1[44:45])












